src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -415,8 +522,27 @@ class WordHighlighter {
 			const myRequestId = ++this.workerRequestTokenId;
 			this.workerRequestCompleted = false;
 
-			this.workerRequest = computeOccurencesAtPosition(this.providers, this.model, this.editor.getSelection(), this.editor.getOption(EditorOption.wordSeparators));
+			const currentEditors = this.codeEditorService.listCodeEditors();
+			const currentModels: ITextModel[] = [];
+			for (const editor of currentEditors) {
+				const model = editor.getModel();
+				if (model) {
+					currentModels.push(model);
+				}
+			}
 
+			const isNotebookEditor = this.editor.getModel()?.uri.scheme === 'vscode-notebook-cell';
+			const notebookCellModels = currentModels.filter(model => model.uri.scheme === 'vscode-notebook-cell');
```

Comment:
For notebook cell models, do we want to highlight only the cells in the same notebook?